DNN Platform/Library/Entities/Urls/AdvancedUrlRewriter.cs

```diff
@@ -2777,7 +2777,15 @@ private void SecurityCheck(HttpApplication app)
                                 //DNN-9158: prevent SSL Offloading infinite redirects
                                 if (!result.IsSecureConnection && result.IsSSLOffloaded && bestFriendlyNoScheme.StartsWith("https"))
                                 {
-                                    bestFriendlyNoScheme = bestFriendlyNoScheme.Replace("https://", "http://");
+                                    if (bestFriendlyNoScheme.Contains("?"))
+                                    {
+                                        var urlParts = bestFriendlyNoScheme.Split('?');

```

Comment:
```suggestion
                                        var urlParts = bestFriendlyNoScheme.Split(new[] { '?' }, count: 2);
```

We might as well guard against multiple `?` characters, right?

Or maybe it should be something like this:
```
if (!result.IsSecureConnection && result.IsSSLOffloaded && bestFriendlyNoScheme.StartsWith("https://"))
{
    bestFriendlyNoScheme = "http://" + bestFriendlyNoScheme.Substring("https://".Length);
}
```

Reply:
Thanks @bdukes , code updated.